vpr/src/place/analytic_placer.h

```diff
@@ -0,0 +1,249 @@
+ * between each block and its target location in the legalized overlap-free placement. When the mathematical system is
+ * again formulated and solved, the pseudo connections pull blocks towards their target locations, which tends to reduce
+ * overlaps in the placement.
+ *
+ * This process of formulating the system, solving, and legalizing is repeated until sufficiently good placement is
+ * acquired. Currently the stopping criterion is 15 iterations without improvement in total_hpwl.
+ *
+ *
+ * Parameters to tweak & things to try out
+ * =======================================
+ * Currently the QoR and runtime of AP+quench combination is slightly worse than SA. See PR #1504 for comparison.
+ * The following parameters/things can be tweaked to find the best configuration:
+ *
+ * * Stopping criteria			when to stop AP iterations, see (AnalyticPlacer::ap_place())
+ * * PlacerHeapCfg.alpha		anchoring strength of pseudo-connection
+ * * PlacerHeapCfg.beta			overutilization factor (@see CutSpreader::SpreaderRegion.overused())
+ * * PlacerHeapCfg.timingWeight	implement timing in AP (@see AnalyticPlacer::build_equations())
+ * * PlacerHeapCfg.criticality	same as above
+ * * Interaction with SA:
+ * 	 * init_t					Initial temperature of annealer after AP (currently init_t = 0)
+ * 	 * quench inner_num			how much swapping in quenching to attemp
+ * 	 * quench_recompute_limit	frequency of criticality update in quenching to improve quench results
+ *
+ * @cite SimPL
+ * Original analytic placer with cut-spreading legalizing was intended for ASIC design, proposed in SimPL.
+ * SimPL: An Effective Placement Algorithm, Myung-Chul Kim, Dong-Jin Lee and Igor L. Markov
+ * http://www.ece.umich.edu/cse/awards/pdfs/iccad10-simpl.pdf
+ *
+ * @cite HeAP
+ * FPGA adaptation of SimPL, targeting FPGAs with heterogeneous blocks located at discrete locations.
+ * Analytical Placement for Heterogeneous FPGAs, Marcel Gort and Jason H. Anderson
+ * https://janders.eecg.utoronto.ca/pdfs/marcelfpl12.pdf
+ *
+ * @cite nextpnr
+ * An implementation of HeAP, which the cut-spreader and legalizer here is based off of. Implementation details
+ * have been modified for the architecture and netlist specification of VTR, and better performance.
+ * nextpnr -- Next Generation Place and Route,  placer_heap, David Shah <david@symbioticeda.com>
+ * https://github.com/YosysHQ/nextpnr
+ */
+
+#    include "vpr_context.h"
+#    include "timing_place.h"
+#    include "PlacementDelayCalculator.h"
+
+typedef signed long int wirelen_t;
+
+/*
+ * @brief Templated struct for constructing and solving matrix equations in analytic placer
+ * Eigen library is used in EquationSystem::solve()
+ */
+template<typename T>
+struct EquationSystem;
+
+/*
+ * @brief helper function to find the index of macro that contains blk
+ * returns index in placementCtx.pl_macros, -1 if blk not in any macros
+ */
+int imacro(ClusterBlockId blk);
+
+/*
+ * @brief helper fucntion to find the head of macro containing blk
```

Comment:
Explain what the head block is, and why it is important (only one represented in the matrix equation). Typo: function.